### DISCUSSION TITLE
PicoGK v1.2.0

### DISCUSSION THREAD

**Discussion by LinKayser** [leap71/PicoGK#19] [General]
**PicoGK v1.2** adds **support for reading and writing OpenVDB .vdb files**. At this time, the openvdb::GRID_LEVEL_SET data type, which is what we use to represent our Voxels object is supported.

PicoGK v1.2 requires the updated 1.2 runtime, which is automatically installed when using the installers.

The following changes were made:

- **NEW**: Added example on how to create a simple lattice and how to extract slices from it (run **PicoGKExamples. SimpleLatticeAndSliceExtract.Task**)
- **FIX**: Including an empty bounding box to an empty bounding box, made bounding box infinitely large.
- **FIX**: Made ASCII vs. Binary detection of STLs more robust
- **FIX**: Loading ASCII STL now correctly throws a "not implemented" exception.
- **CHANGE**: If loading an STL results in an empty mesh, an exception is thrown
- **CHANGE**: If bounding box of viewer objects is empty, matrix calculations are skipped, to avoid invalid math operations.
- **NEW**: Added function to append one mesh to another (no intelligence, just adds the triangles, regardless of the mesh topology). Useful for helper meshes in the GUI, for example.
- **NEW**: Added way to create mesh from BBox3
- **FIX**: Removed unimplemented Voxels.BoolAddSmooth function, which was throwing an error.
- **NEW**: Added TempFolder class to hold a temp folder that is automatically created and deleted when no longer used.
- **FIX**: Width/Height were flipped in GetVoxelSlice, if you worked around by flipping x/y, you need to undo that change. Now works correctly: Width is X, Height is Y
- **FIX:** Fixed NaN values at the center of flat cap beams of lattices. Flat cap beam lattices now no longer have a tiny hole in their middle
- **FIX**: Memory of meshes was never freed in the PicoGK Runtime
- **NEW**: Added a function to wait for the creation of a file
- **NEW**: Support for reading and writing OpenVDB .VDB files. (For details, run **PicoGKExamples. OpenVdbExample.Task**)
- **CHANGE**: PicoGK Example, which is installed by the installers incorporates the latest LEAP 71 Libraries (Shape Kernel, Lattice Library).

<hr /><em>This discussion was created from the release <a href='https://github.com/leap71/PicoGK/releases/tag/PicoGK-v1.2.0'>PicoGK v1.2.0</a>.</em>

**Comment by LinKayser** [comment]
Dear all, I just pushed the latest release. Biggest change is OpenVDB format support, which you may found helpful in interop scenarios. Much more work to do, for example, right now, it doesn't store the physical dimensions of the voxel field (in mm), so your physical sizes when reloading the VDB will change when your library voxel size changes.

I have plans to address this in the future. Also, we are working on supporting OpenVDB fog volumes and read them into a generic scalar or vector field. This will allow us to more effectively interact with simulation tools (and incorporate a simulation tool's results into the algorithms, by reading a field of float scalars or 3D vectors.

Thanks for everyone who gave feedback (and created pull requests).

Lin

**Reply by sambeyene** [reply]
Hi LinKayser,

I installed PicoGK and can see all files/folders including Examples.  However, the .exe file, or icon is no where to be found.  
Could you please help on how to find and execute the software.

Thanks
Sam

**Reply by LinKayser** [reply]
Dear Sam — you interact with the software through Microsoft Visual Studio. Check out the Getting Started section in the documentation: https://github.com/leap71/PicoGK/tree/main/Documentation
